src/Confluent.Kafka/Producer.cs

```diff
@@ -635,93 +957,364 @@ public class Producer<TKey, TValue> : ISerializingProducer<TKey, TValue>, IDispo
+        /// </remarks>
         public void ProduceAsync(string topic, TKey key, TValue val, IDeliveryHandler<TKey, TValue> deliveryHandler)
             => serializingProducer.ProduceAsync(topic, key, val, deliveryHandler);
 
+        /// <summary>
+        ///     Asynchronously send a single message to the broker (order of delivery reports strictly guarenteed).
+        /// </summary>
+        /// <remarks>
+        ///     Notification of delivery reports is via an IDeliveryHandler instance. Use IDeliveryHandler variants of 
+        ///     ProduceAsync if you require notification of delivery reports strictly in the order they were 
+        ///     acknowledged by the broker / failed (failure may be via broker or local).
+        ///     
+        ///     Refer to <see cref="ProduceAsync(string, TKey, TValue, int, bool)" />
+        ///     for more information.
+        /// </remarks>
         public void ProduceAsync(string topic, TKey key, TValue val, int partition, bool blockIfQueueFull, IDeliveryHandler<TKey, TValue> deliveryHandler)
             => serializingProducer.ProduceAsync(topic, key, val, partition, blockIfQueueFull, deliveryHandler);
 
+        /// <summary>
+        ///     Asynchronously send a single message to the broker (order of delivery reports strictly guarenteed).
+        ///     See <see cref="ProduceAsync(string, TKey, TValue, int, bool, IDeliveryHandler{TKey, TValue})"/> for more information.
+        /// </summary>
+        /// <remarks>
+        ///     Blocks if the send queue is full. Note: if background polling is disabled and Poll is
+        ///     not being called in another thread, this will block indefinitely.
+        /// </remarks>
         public void ProduceAsync(string topic, TKey key, TValue val, int partition, IDeliveryHandler<TKey, TValue> deliveryHandler)
             => serializingProducer.ProduceAsync(topic, key, val, partition, deliveryHandler);
 
+        /// <summary>
+        ///     Asynchronously send a single message to the broker (order of delivery reports strictly guarenteed).
+        ///     See <see cref="ProduceAsync(string, TKey, TValue, int, bool, IDeliveryHandler{TKey, TValue})"/> for more information.
+        /// </summary>
+        /// <remarks>
+        ///     The partition the message is produced to is determined using the configured partitioner.
+        /// </remarks>
         public void ProduceAsync(string topic, TKey key, TValue val, bool blockIfQueueFull, IDeliveryHandler<TKey, TValue> deliveryHandler)
             => serializingProducer.ProduceAsync(topic, key, val, blockIfQueueFull, deliveryHandler);
 
 
+        /// <summary>
+        ///     Raised when there is information that should be logged.
+        /// </summary>
+        /// <remarks>
+        ///     You can specify the log level with the 'log_level' configuration
+        ///     property. You also probably want to specify one or more debug
```

Comment:
changed this to still note log_level and debug, but dissuade people from using it.